core/src/main/java/com/linecorp/armeria/client/metric/MetricCollectingClientBuilder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.metric;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.common.metric.AbstractMetricCollectingBuilder;
+import com.linecorp.armeria.common.metric.MeterIdPrefixFunction;
+
+/**
+ * Builds a {@link MetricCollectingClient} instance.
+ */
+public final class MetricCollectingClientBuilder
+        extends AbstractMetricCollectingBuilder<MetricCollectingClient, HttpClient> {
+
+    MetricCollectingClientBuilder(MeterIdPrefixFunction meterIdPrefixFunction) {
+        super(meterIdPrefixFunction);
+    }
+
```

Comment:
I realized that we still need to override the setter in order to represent it properly in the Javadoc reference. So we can do:
```java
@Override
public MetricCollectingClientBuilder successFunction(...) {
    return (MetricCollectingClientBuilder) super.successFunction(...)
}
```